artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/DefaultSenderController.java

```diff
@@ -207,6 +212,10 @@ public Consumer init(ProtonServerSenderContext senderContext) throws Exception {
          }
          source.setAddress(queue.toString());
       } else {
+         final String sourceAddress = ParameterisedAddress.extractAddress(source.getAddress());
+
+         addressParameters = ParameterisedAddress.extractParameters(source.getAddress());
```

Comment:
The method cant handle being passed a null, but its possible this value is null.

I'd guess that might even explain the unit test failure (from an exception type having changed, to an internal exception)